front/src/applications/editor/tools/rangeEdition/components/RangeEditionLeftPanel.tsx

```diff
@@ -53,6 +72,66 @@ const RangeEditionLeftPanel = () => {
     });
   };
 
+  const unselectSwitch = (sw: string) => () => {
+    setState({
+      selectedSwitchs: selectedSwitchs.filter((s) => s !== sw),
+    });
+  };
+
+  const searchRoutesFromSwitch = async () => {
+    const body = selectedSwitchs.reduce(
+      (accumulator, value) => ({ ...accumulator, [value]: null }),
+      {}
+    );
+    setState({
+      optionsState: { type: 'loading' },
+    });
+    const routesCandidate = await getRoutesFromSwitch({
+      infraId: infraID as number,
+      body,
+    }).unwrap();
+    setSwitchsRoutesCandidate(routesCandidate as string[]);
+    setState({
+      optionsState: { type: 'idle' },
+    });
+  };
+
+  const getTrackRanges = async (routes: string) => {
```

Comment:
```suggestion
  const getTrackRangesFromRouteId = async (routeId: string) => {
```

Reply:
Can't change or it will need a reassignation just after because of the osrdEditoastApi arg requirements, but change the function signature

Reply:
done